### PR TITLE
Update project libraries

### PR DESCRIPTION
**Pull request by vshkl**
* Update project libraries
* Migrate project to RxJava2
* Migrate project to Room
* Use `IntDef` annotation
* Some code reafactoring

### CODE REVIEW

File: app/src/main/java/com/datarockets/mnchkn/data/local/DatabaseHelper.kt [hunk 1]
```diff
@@ -1,296 +1,105 @@
...
 
-    val briteDb: BriteDatabase
-
-    init {
-        val briteBuilder = SqlBrite.Builder()
-                .logger { message -> Timber.tag("Database").v(message) }
-        briteDb = briteBuilder.build().wrapDatabaseHelper(dbOpenHelper, Schedulers.immediate())
-        briteDb.setLoggingEnabled(true)
+    fun setPlayer(player: Player): Single<Player> = Single.create {
+        val playerId = MunchkinApplication.database.playerDao().setPlayer(PlayerEntityMapper.transform(player))
```
Comment (chyrta) [inline app/src/main/java/com/datarockets/mnchkn/data/local/DatabaseHelper.kt]: @vshkl What about having private finalized `playerDao` and `gameDao` in database helper which you can pass as dependencies with Dagger? Just create methods `providesGameDao` and `providePlayerDao` in ApplicationModule. And pass `gameDao` and `playerDao` via constructor of database helper. Personally I don't like such a long call like `MunchkinApplication.database.playerDao` and bla-bla-bla.

File: app/src/main/java/com/datarockets/mnchkn/injection/modules/ApplicationModule.kt [hunk 2]
```diff
@@ -5,25 +5,22 @@ import android.content.Context
...
 
 @Module
-class ApplicationModule(protected val mApplication: Application) {
+class ApplicationModule(private val application: Application) {
 
     @Provides
     fun providesApplication(): Application {
-        return mApplication
+        return application
     }
```
Comment (chyrta) [inline app/src/main/java/com/datarockets/mnchkn/injection/modules/ApplicationModule.kt:25]: @vshkl This is the place where you can initialize RoomDatabase and then pass the result as a parameter for providing `gameDao` and `playerDao`. Please, consider about doing this.

**Review by chyrta — CHANGES_REQUESTED** [review]

**Comment by vshkl** [comment]
@chyrta, update as per your suggestions.

File: app/src/main/java/com/datarockets/mnchkn/injection/modules/ApplicationModule.kt [hunk 3]
```diff
@@ -36,4 +47,22 @@ class ApplicationModule(private val application: Application) {
                 .core(crashlyticsCore)
                 .build()
     }
+
+    @Singleton
+    @Provides
+    fun providesDatabase(application: Application): MunchkinDatabase {
```
Comment (chyrta) [inline app/src/main/java/com/datarockets/mnchkn/injection/modules/ApplicationModule.kt]: You need to pass Context there, not an Application. Actually, it looks great.

**Review by chyrta — CHANGES_REQUESTED** [review]

**Comment by vshkl** [comment]
@chyrta, done. Provide database passing applicationContext.

**Comment by chyrta** [comment]
@vshkl Approved 👍

**Pull request merged**